app/src/main/java/com/toolbartest/RecyclerAdapter.java

```diff
@@ -46,17 +45,22 @@ public int getItemCount() {
         return mData.size();
     }
 
-    class MyViewHolder extends RecyclerView.ViewHolder {
+    class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
 
         TextView title;
         ImageView listIcon;
 
-        public MyViewHolder(View itemView,
-                            View.OnClickListener itemClickListener) {
+        public MyViewHolder(View itemView) {
             super(itemView);
             title = (TextView)itemView.findViewById(R.id.listText);
             listIcon = (ImageView)itemView.findViewById(R.id.listIcon);
-            itemView.setOnClickListener(itemClickListener);
+            itemView.setOnClickListener(this);
+        }
+
+        @Override
+        public void onClick(View v) {
+                MainActivity mainActivity = (MainActivity) context;
```

Comment:
This adds dependency on MainActivity, making it difficult (or impossible) to drop the fragment in any other Activity. Callbacks doesn't have such dependencies.